test/unit-test/test_lib/test_analysis/test_user.py

```diff
@@ -1,34 +1,138 @@
 import unittest
 from lib.analysis import user
+import lib.config as config
+from lib.in_out import reader, saver
+import lib.nickTracker as nickTracker
+from mock import patch
+import networkx as nx
 import lib.util as util
 import os
+import StringIO
+import sys
+import filecmp
 
 current_directory = os.path.dirname(os.path.realpath(__file__))
+state0 = 0
+state1 = 0
+def build_graphs_util(nick_sender, nick_receiver, time, year, month, day, day_graph, aggr_graph):
+    build_graphs_list = util.load_from_disk(current_directory + "/data/user_test/build_graphs_list")
+    day_graph = build_graphs_list[state][0]
+    aggr_graph = build_graphs_list[state][1]
+    state0 += 1 
+def rec_list_splice_util(rec_list):
+    rec_list_splice_list = util.load_from_disk(current_directory + "/data/user_test/rec_list_splice_list")
+    rec_list[2] = rec_list_splice_list[state1]
+    state1 += 1
 
 class UserTest(unittest.TestCase):
 
     def setUp(self):
-        self.log_data = util.load_from_disk(current_directory+ "/data/log_data")
-        self.nicks_hash = util.load_from_disk(current_directory+ "/data/nicks_hash")
+        #./data/user_test/log_data has data data for the following configuraton:
+        #   STARTING_DATE = "2013-2-1"
+        #   ENDING_DATE = "2013-2-15"
+        #   CHANNEL_NAME = ["#kubuntu-devel"] 
+        self.log_data = util.load_from_disk(current_directory+"/data/user_test/log_data")
+        self.nicks_hash = util.load_from_disk(current_directory+ "/data/user_test/nicks_hash")
+        self.nick_same_list = util.load_from_disk(current_directory+"/data/user_test/nick_same_list")
+        state0 = 0
+        state1 = 0
 
     def tearDown(self):
         self.log_data = None
         self.nicks_hash = None
+        self.nick_same_list = None
 
-    def test_nick_change_graph(self):
-        pass
+    @patch("util.splice_find",  autospec = True)
+    @patch("util.build_graphs",  autospec = True)
+    @patch("util.get_year_month_day",  autospec = True)
+    def test_nick_change_graph(self, mock_get_year_month_day,mock_build_graphs, mock_splice_find):
+        mock_splice_find.side_effect = util.load_from_disk(current_directory + "/data/user_test/nick_change_graph/splice_find_list")
+        mock_build_graphs.side_effect = build_graphs_util 
+        nick_change_graph_list = user.nick_change_graph(self.log_data, True)
+        expected_nick_change_graph_list = util.load_from_disk(current_directory + "/data/user_test/nick_change_graph_list")
+        for i in range(len(nick_change_graph_list)):
```

Comment:
Conisder using [xrange](https://stackoverflow.com/questions/94935/what-is-the-difference-between-range-and-xrange-functions-in-python-2-x).